File: planner/app/src/main/java/com/pro3/planner/dialogs/VisibilityDialog.java
package com.pro3.planner.dialogs;

import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.WindowManager;

import com.pro3.planner.R;
import com.pro3.planner.adapters.VisibilityPagerAdapter;

/**
 * Created by linus_000 on 17.11.2016.
 */

public class VisibilityDialog extends android.support.v4.app.DialogFragment {

    FragmentPagerAdapter adapterViewPager;



    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        Dialog dialog = super.onCreateDialog(savedInstanceState);

        WindowManager.LayoutParams layoutParams = new WindowManager.LayoutParams();
        layoutParams.copyFrom(dialog.getWindow().getAttributes());
        layoutParams.height = ViewGroup.LayoutParams.WRAP_CONTENT;
        layoutParams.width = ViewGroup.LayoutParams.MATCH_PARENT;

        dialog.getWindow().setAttributes(layoutParams);

        return dialog;
    }

    @Override
    public void onResume() {
        super.onResume();


        WindowManager.LayoutParams layoutParams = new WindowManager.LayoutParams();
        layoutParams.copyFrom(getDialog().getWindow().getAttributes());
        layoutParams.height = ViewGroup.LayoutParams.WRAP_CONTENT;
        layoutParams.width = ViewGroup.LayoutParams.MATCH_PARENT;

        getDialog().getWindow().setAttributes(layoutParams);

    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {

        adapterViewPager = new VisibilityPagerAdapter(getChildFragmentManager(), getActivity());

        View content = inflater.inflate(R.layout.alertdialog_body_visibility, container);
        ViewPager vpPager = (ViewPager) content.findViewById(R.id.menu_visibility_pager);
        vpPager.setAdapter(adapterViewPager);

        TabLayout tabLayout = (TabLayout) content.findViewById(R.id.sliding_tabs);
        tabLayout.setupWithViewPager(vpPager);

        return content;
    }
}

File: planner/app/src/main/java/com/sunilson/firenote/presentation/shared/dialogs/visibilityDialog/VisibilityDialog.kt
package com.sunilson.firenote.presentation.shared.dialogs.visibilityDialog

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import com.google.android.material.tabs.TabLayout
import androidx.fragment.app.Fragment
import androidx.viewpager.widget.ViewPager
import android.view.*
import com.sunilson.firenote.R
import com.sunilson.firenote.presentation.shared.base.BaseDialogFragment
import com.sunilson.firenote.presentation.shared.dialogs.visibilityDialog.adapters.VisibilityPagerDapter
import dagger.android.AndroidInjector
import dagger.android.DispatchingAndroidInjector
import dagger.android.support.AndroidSupportInjection
import dagger.android.support.HasSupportFragmentInjector
import javax.inject.Inject

class VisibilityDialog : BaseDialogFragment(), HasSupportFragmentInjector {

    @Inject
    lateinit var dispatchingAndroidInjector: DispatchingAndroidInjector<Fragment>

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val dialog = super.onCreateDialog(savedInstanceState)
        setDimensions(dialog)
        dialog.requestWindowFeature(Window.FEATURE_NO_TITLE)
        return dialog
    }

    override fun onResume() {
        super.onResume()
        setDimensions(requireDialog())
    }

    private fun setDimensions(dialog: Dialog) {
        val layoutParams = WindowManager.LayoutParams()
        layoutParams.copyFrom(dialog.window?.attributes)
        layoutParams.height = ViewGroup.LayoutParams.WRAP_CONTENT
        layoutParams.width = ViewGroup.LayoutParams.WRAP_CONTENT
        dialog.window?.attributes = layoutParams
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val adapterViewPager = VisibilityPagerDapter(childFragmentManager, activity as Context)
        val content = inflater.inflate(R.layout.alertdialog_body_visibility, container)
        val vpPager = content.findViewById<ViewPager>(R.id.menu_visibility_pager)
        vpPager.adapter = adapterViewPager
        val tabLayout = content.findViewById<TabLayout>(R.id.sliding_tabs)
        tabLayout.setupWithViewPager(vpPager)
        return content
    }

    override fun onAttach(context: Context) {
        AndroidSupportInjection.inject(this)
        super.onAttach(context)
    }

    override fun toggleLoading(loading: Boolean, message: String?) {}

    override fun supportFragmentInjector(): AndroidInjector<Fragment> = dispatchingAndroidInjector

    companion object {
        fun newInstance(): VisibilityDialog {
            return VisibilityDialog()
        }
    }
}